stock_inventory/models/stock_inventory.py

```diff
@@ -0,0 +1,193 @@
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class InventoryAdjustmentsGroup(models.Model):
+    _name = "stock.inventory"
+    _description = "Inventory Adjustment Group"
+    _order = "date desc, id desc"
+
+    name = fields.Char(
+        required=True, default="Inventory ", string="Inventory Reference"
+    )
+
+    date = fields.Datetime(default=lambda self: fields.Datetime.now())
+
+    state = fields.Selection(
+        [("draft", "Draft"), ("in_progress", "In Progress"), ("done", "Done")],
+        default="draft",
+    )
+
+    location_ids = fields.Many2many(
+        "stock.location", string="Location", domain="[('usage', '=', 'internal')]"
+    )
+
+    product_selection = fields.Selection(
+        [
+            ("all", "All Products"),
+            ("manual", "Manual Selection"),
+            ("category", "Product Category"),
+            ("one", "One Product"),
+            ("lot", "Lot Serial Number"),
+        ],
+        default="all",
+        required=True,
+    )
+
+    product_ids = fields.Many2many("product.product", string="Products")
+
+    stock_quant_ids = fields.Many2many("stock.quant", string="Inventory Adjustment")
+
+    category_id = fields.Many2one("product.category", string="Product Category")
+
+    lot_ids = fields.Many2many(
+        "stock.production.lot",
+        string="Lot/Serial Number",
+    )
+
+    stock_move_ids = fields.One2many(
+        "stock.move.line",
+        "inventory_adjustment_id",
+        string="Inventory Adjustments Done",
+    )
+
+    count_stock_quants = fields.Integer(
+        compute="_compute_count_stock_quants", string="Adjustments"
+    )
+
+    count_stock_quants_string = fields.Char(
+        compute="_compute_count_stock_quants", string="Adjustments"
+    )
+
+    count_stock_moves = fields.Integer(
+        compute="_compute_count_stock_moves", string="Stock Moves Lines"
+    )
+
+    @api.depends("stock_quant_ids")
+    def _compute_count_stock_quants(self):
+        self.count_stock_quants = len(self.stock_quant_ids)
+        count_todo = len(
+            self.stock_quant_ids.search(
+                [("id", "in", self.stock_quant_ids.ids), ("to_do", "=", "True")]
+            )
+        )
+        self.count_stock_quants_string = "{} / {}".format(
+            count_todo, self.count_stock_quants
+        )
+
+    @api.depends("stock_move_ids")
+    def _compute_count_stock_moves(self):
+        sm_ids = self.mapped("stock_move_ids").ids
+        self.count_stock_moves = len(sm_ids)
+
+    def action_state_to_in_progress(self):
+        active_rec = self.env["stock.inventory"].search([("state", "=", "in_progress")])
+        if active_rec:
```

Comment:
Should we extend this? Checking if there are counts with same product or same lots or same locations? Counting two locations at the same time should be possible?

Reply:
I don't think so because you can only do an Adjustment at a time, so repeated counts are "imposible" (unless something magic happens).